### PR TITLE
Proxies sampling documentation

### PR DESCRIPTION
**Pull request by dussault-antoine**
### What does this PR do?

Add documentation for sampling in proxies

### Motivation

Missing docs for these specific cases. Many customers struggling (see https://docs.google.com/document/d/1R2VE24kzoIS8J32SjfcmUr4clsMQvT2DLn5euRrz71E/edit#)

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/tracing/setup_overview/proxy_setup/_index.md [hunk 1]
```diff
@@ -211,9 +211,71 @@ stats_config:
...
+
+You can also define an explicit sampling rate between `0.0` (0%) and `1.0` (100%) by service. For instance, to set the sample rate to 10% for service `envoy-proxy`, set :
+
+```
+DD_TRACE_SAMPLING_RULES=[{"service": "envoy-proxy","sample_rate": 0.1}]
+```
+
+### How to apply the DD_TRACE_SAMPLING_RULES configuration option ?
+
+If envoy is executed by a shell script, set the environment variable right before executing envoy in the shell script
```
[the remark this answers is not in the document]
Comment (dussault-antoine) [reply content/en/tracing/setup_overview/proxy_setup/_index.md:235]: each proxy type is already a tab, we cannot make sub tabs i think :/

[the remark this answers is not in the document]
Comment (kayayarai) [reply content/en/tracing/setup_overview/proxy_setup/_index.md:237]: correct, we cannot

File: content/en/tracing/setup_overview/proxy_setup/_index.md [hunk 2]
```diff
@@ -232,7 +299,8 @@ The available [environment variables][2] depend on the version of the C++ tracer
 | v1.9 | v0.3.6 |
 
 [1]: https://github.com/DataDog/dd-opentracing-cpp/tree/master/examples/envoy-tracing
-[2]: /tracing/setup/cpp/#environment-variables
+[2]: tracing/trace_ingestion/mechanisms#in-the-agent
```
Comment (kayayarai) [inline content/en/tracing/setup_overview/proxy_setup/_index.md:302]:
```suggestion
[2]: /tracing/trace_ingestion/mechanisms#in-the-agent
```

File: content/en/tracing/setup_overview/proxy_setup/_index.md [hunk 3]
```diff
@@ -211,9 +211,76 @@ stats_config:
       - prefix: "cluster.datadog_agent."
 ```
 
+## Sampling
+
+The sampling rate for traces starting from Envoy can be set between `0.0` (0%) and `1.0` (100%).
+
+Configure the sampling rate in order to control the volume of traces sent to Datadog by using the parameter `DD_TRACE_SAMPLING_RULES`.
+By default, 100% of traces starting from Envoy are sampled.
```
Comment (kayayarai) [inline content/en/tracing/setup_overview/proxy_setup/_index.md:219]:
```suggestion
To control the volume of traces starting from Envoy that are sent to Datadog, specify a sampling rate by setting the parameter `DD_TRACE_SAMPLING_RULES` to a value between `0.0` (0%) and `1.0` (100%). If no value is specified, 100% of traces starting from Envoy are sent.
```

File: content/en/tracing/setup_overview/proxy_setup/_index.md [hunk 4]
```diff
@@ -211,9 +211,76 @@ stats_config:
...
+
+To use the [Datadog Agent calculated sampling rates][2] (10 traces per second per Agent) and ignore the default sampling rule set to 100%, set the parameter `DD_TRACE_SAMPLING_RULES` to an empty array:
+
+```
+DD_TRACE_SAMPLING_RULES=[]
+```
+
+You can also define an explicit sampling rate between `0.0` (0%) and `1.0` (100%) by service. For instance, to set the sample rate to 10% for service `envoy-proxy`, set :
```
Comment (kayayarai) [inline content/en/tracing/setup_overview/proxy_setup/_index.md:227]:
```suggestion
You can also define an explicit sampling rate between `0.0` (0%) and `1.0` (100%) by service. For example, to set the sample rate to 10% for service `envoy-proxy`:
```

File: content/en/tracing/setup_overview/proxy_setup/_index.md [hunk 5]
```diff
@@ -211,9 +211,76 @@ stats_config:
...
+
+```
+DD_TRACE_SAMPLING_RULES=[{"service": "envoy-proxy","sample_rate": 0.1}]
+```
+
+### How to apply the DD_TRACE_SAMPLING_RULES configuration option ?
```
Comment (kayayarai) [inline content/en/tracing/setup_overview/proxy_setup/_index.md:233]:
```suggestion
### Setting the `DD_TRACE_SAMPLING_RULES` configuration option
```

File: content/en/tracing/setup_overview/proxy_setup/_index.md [hunk 6]
```diff
@@ -211,9 +211,76 @@ stats_config:
...
+
```
Comment (kayayarai) [inline content/en/tracing/setup_overview/proxy_setup/_index.md:234]:
```suggestion

Use one of the following methods to configure your sampling rate, depending on how you run Envoy:

```

File: content/en/tracing/setup_overview/proxy_setup/_index.md [hunk 7]
```diff
@@ -211,9 +211,76 @@ stats_config:
...
+#### In a Shell script
+
+If envoy is executed by a shell script, set the environment variable right before executing envoy in the shell script
```
Comment (kayayarai) [inline content/en/tracing/setup_overview/proxy_setup/_index.md:237]:
```suggestion
- **By shell script**: Set the environment variable immediately before executing `envoy` in the script:
```

File: content/en/tracing/setup_overview/proxy_setup/_index.md [hunk 8]
```diff
@@ -211,9 +211,76 @@ stats_config:
...
+
+```
+#!/bin/sh
+export DD_TRACE_SAMPLING_RULES=[]
+envoy -c envoy-config.yaml
+```
+#### Docker compose setup
+
+If envoy is executed as a service in a docker-compose setup, set the environment variable in the `environment` section of the service definition:
```
Comment (kayayarai) [inline content/en/tracing/setup_overview/proxy_setup/_index.md:246]:
```suggestion

- **In a Docker Compose setup**: Set the environment variable in the `environment` section of the service definition:
```

File: content/en/tracing/setup_overview/proxy_setup/_index.md [hunk 9]
```diff
@@ -211,9 +211,76 @@ stats_config:
...
+        - /etc/envoy/envoy.yaml
+    volumes:
+        - './envoy.yaml:/etc/envoy/envoy.yaml:ro'
+    environment:
+        - DD_TRACE_SAMPLING_RULES=[]
+```
+
+#### Kubernetes pod container
+
+If envoy is a container inside of a Kubernetes pod, speficy the env variable in the `env` section of the relevant `containers` entry of the pod's spec:
```
Comment (kayayarai) [inline content/en/tracing/setup_overview/proxy_setup/_index.md:265]:
```suggestion
- **As a container inside a Kubernetes pod**: specify the environment variable in the `env` section of the corresponding `containers` entry of the pod's spec:
```

File: content/en/tracing/setup_overview/proxy_setup/_index.md [hunk 10]
```diff
@@ -371,6 +439,39 @@ To set a different service name per Ingress using annotations:
 ```
 The above overrides the default `nginx-ingress-controller.ingress-nginx` service name.
 
+### Sampling
+
+The sampling rate for traces starting from Nginx services can be set between `0.0` (0%) and `1.0` (100%).
+
+Configure the sampling rate in order to control the volume of traces sent to Datadog by using the parameter `DD_TRACE_SAMPLING_RULES`.
+By default, 100% of traces starting from Nginx are sampled (the Kubernetes Nginx ingress controller uses the [v1.2.1][8] of the `dd-opentracing-cpp` library).
```
Comment (kayayarai) [inline content/en/tracing/setup_overview/proxy_setup/_index.md:447]:
```suggestion
To control the volume of traces starting from Nginx that are sent to Datadog, specify a sampling rate by setting the parameter `DD_TRACE_SAMPLING_RULES` to a value between `0.0` (0%) and `1.0` (100%). If no value is specified, 100% of traces starting from Nginx are sent. The Kubernetes Nginx ingress controller uses [v1.2.1][8] of the `dd-opentracing-cpp` library).
```

File: content/en/tracing/setup_overview/proxy_setup/_index.md [hunk 11]
```diff
@@ -371,6 +439,39 @@ To set a different service name per Ingress using annotations:
...
+
+To use the [Datadog Agent calculated sampling rates][9] (10 traces per second per Agent) and ignore the default sampling rule set to 100%, set the parameter `DD_TRACE_SAMPLING_RULES` to an empty array:
+
+```
+DD_TRACE_SAMPLING_RULES=[]
+```
+
+Append a `main-snippet` data entry to the controller's ConfigMap, as below:
```
Comment (kayayarai) [inline content/en/tracing/setup_overview/proxy_setup/_index.md:455]:
```suggestion
To configure `DD_TRACE_SAMPLING_RULES`, add a `main-snippet` entry in the `data` section of the controller's ConfigMap:
```

File: content/en/tracing/setup_overview/proxy_setup/_index.md [hunk 12]
```diff
@@ -436,9 +539,35 @@ template:
 For [CronJobs][8], the `app` label should be added to the job template, as the generated name comes from the `Job` instead
 of the higher-level `CronJob`.
 
+### Sampling
+
+The sampling rate for traces starting from Istio services can be set between `0.0` (0%) and `1.0` (100%).
+
+Configure the sampling rate in order to control the volume of traces sent to Datadog by using the parameter `DD_TRACE_SAMPLING_RULES`.
+By default, 100% of traces starting from Istio are sampled.
```
Comment (kayayarai) [inline content/en/tracing/setup_overview/proxy_setup/_index.md:547]:
```suggestion
To control the volume of traces starting from Istio that are sent to Datadog, specify a sampling rate by setting the parameter `DD_TRACE_SAMPLING_RULES` to a value between `0.0` (0%) and `1.0` (100%). If no value is specified, 100% of traces starting from Istio are sent. 
```

File: content/en/tracing/setup_overview/proxy_setup/_index.md [hunk 13]
```diff
@@ -436,9 +539,35 @@ template:
...
+
+To use the [Datadog Agent calculated sampling rates][9] (10 traces per second per Agent) and ignore the default sampling rule set to 100%, set the parameter `DD_TRACE_SAMPLING_RULES` to an empty array:
+
+```
+DD_TRACE_SAMPLING_RULES=[]
+```
+
+In each deployment whose namespace is labeled istio-injection=enabled, the environment variable can be set as part of the apm.datadoghq.com/env annotation of the deployment spec template.
```
Comment (kayayarai) [inline content/en/tracing/setup_overview/proxy_setup/_index.md:555]:
```suggestion
To configure `DD_TRACE_SAMPLING_RULES`, in each deployment whose namespace is labeled `istio-injection=enabled`, set the environment variable as part of the `apm.datadoghq.com/env` annotation of the deployment spec template:
```

**Review by kayayarai — CHANGES_REQUESTED** [review]

**Review by kayayarai — APPROVED** [review]

**Pull request merged**